sw/host/hsmtool/src/util/shredder.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use rand::prelude::*;
+use std::ops::{Deref, DerefMut};
+
+/// The Shreddable trait implements a method to erase an object's memory before drop.
+pub trait Shreddable {
+    fn shred(&mut self);
+}
+
+/// Shredder holds an item and erases it upon drop.
+pub struct Shredder<T: Shreddable>(pub T);
+
+impl<T: Shreddable> Deref for Shredder<T> {
+    type Target = T;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<T: Shreddable> DerefMut for Shredder<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl<T: Shreddable> Drop for Shredder<T> {
+    fn drop(&mut self) {
+        self.0.shred();
+    }
+}
+
+impl Shreddable for String {
+    fn shred(&mut self) {
+        let mut rng = thread_rng();
+        // SAFETY: We are clobbering the string value with random data,
+        // but we immediately truncate the string to length zero.
+        //
+        // In the typical case, we drop the string right after shredding.
+        unsafe {
+            for v in self.as_mut_vec().iter_mut() {
+                *v = rng.gen();
+            }
+        }
+        self.truncate(0);
```

Comment:
I deleted the Shreddable trait and opted to use the `zeroize` crate.  The zeroize crate specifically claims that it will zero out data without being defeated by the optimizer.